webapp/src/ts/modals/training-cards/training-cards.component.ts

```diff
@@ -0,0 +1,217 @@
+    // see https://github.com/medic/cht-core/issues/2198#issuecomment-210202785 for AngularJS behavior
+    this.enketoService.unload(this.form);
+    this.globalActions.clearEnketoStatus();
+  }
+
+  private loadForm() {
+    return this.ngZone.runOutsideAngular(() => this._loadForm());
+  }
+
+  private async _loadForm() {
+    try {
+      this.loadingContent = true;
+      this.hideModalFooter = true;
+      this.geoHandle && this.geoHandle.cancel();
+      this.geoHandle = this.geolocationService.init();
+      const form = await this.xmlFormsService.get(this.trainingForm);
+      return this.ngZone.run(() => this.renderForm(form));
+    } catch(error) {
+      this.setError();
+      console.error('Error fetching form.', error);
+    }
+  }
+
+  private async renderForm(form) {
+    try {
+      const selector = `#${this.formWrapperId}`;
+      this.form = await this.enketoService.render(selector, form, undefined, undefined, this.resetFormError.bind(this));
+      this.loadingContent = false;
+      this.recordTelemetryPostRender();
+    } catch(error) {
+      this.setError();
+      console.error('Error rendering form.', error);
+    }
+  }
+
+  private subscribeToStore() {
+    const reduxSubscription = combineLatest(
+      this.store.select(Selectors.getEnketoStatus),
+      this.store.select(Selectors.getEnketoSavingStatus),
+      this.store.select(Selectors.getEnketoError),
+      this.store.select(Selectors.getTrainingCard),
+    ).subscribe(([
+      enketoStatus,
+      enketoSaving,
+      enketoError,
+      trainingForm,
+    ]) => {
+      this.enketoStatus = enketoStatus;
+      this.enketoSaving = enketoSaving;
+      this.enketoError = enketoError;
+
+      if (trainingForm && trainingForm !== this.trainingForm) {
+        this.trainingForm = trainingForm;
+        this.loadForm();
+      }
+    });
+    this.subscription.add(reduxSubscription);
+  }
+
+  private reset() {
+    this.resetFormError();
+    this.contentError = false;
+    this.trainingForm = null;
+    this.form = null;
+    this.loadingContent = true;
+    this.hideModalFooter = true;
+  }
+
+  private resetFormError() {
+    if (this.enketoError) {
+      this.globalActions.setEnketoError(null);
+    }
+  }
+
+  setError() {
+    this.errorTranslationKey = 'training_cards.error.loading';
+    this.loadingContent = false;
+    this.hideModalFooter = false;
+    this.contentError = true;
+  }
+
+  async saveForm() {
+    if (this.enketoSaving) {
+      console.debug('Attempted to call TrainingCardsComponent:saveForm more than once');
+      return;
+    }
+
+    this.recordTelemetryPreSave();
+    this.globalActions.setEnketoSavingStatus(true);
+    this.resetFormError();
+
+    try {
+      const docs = await this.enketoService.save(this.trainingForm, this.form, this.geoHandle);
+      console.debug('Saved form and associated docs', docs);
+      this.globalActions.setSnackbarContent(this.translateService.instant('training_cards.form.saved'));
```

Comment:
Don't use `instant` if you can help it. It's better to use the async version and just show the snackbar once translation is returned. The reason is in case the translation service is not ready yet.